util/version/version.go

```diff
@@ -162,14 +162,13 @@ func (v buildIdentifier) compare(o buildIdentifier) int {
 			return 1
 		}
 	} else { // both are strings
-		switch {
-		case v.IdentifierStr < o.IdentifierStr:
-			return -1
-		case v.IdentifierStr == o.IdentifierStr:
+		// In order to support random build indentifiers, like commit hashes,
+		// always return -1 when the strings are different to always signal
+		// there is version change
+		if v.IdentifierStr == o.IdentifierStr {
 			return 0
-		default:
-			return 1
 		}
+		return -1
```

Comment:
What about returning 2 in this case, indicated that things are different (given that -1 indicates that  -1 == v is less than o, which is not correct)

Reply:
Note: this requires a change in the `isUpgrading` func in`controllers/topology/contract/controlplane.go`, from 

```
return version.Compare(specV, statusV, version.WithBuildTags()) == 1, nil
```

to

```
// NOTE: we are considering the control plane upgrading when the version is greater
// or when the version has a different build metadata.
return version.Compare(specV, statusV, version.WithBuildTags()) >= 1, nil
```

Reply:
I like it, done